### ISSUE TITLE
FireAndForget delivery combined with deactivating a grain causes an error on success due to confirmation delivery attempt

### ISSUE THREAD

**Issue by highlyunavailable** [dotnet/orleans#7029]
I'm getting the following log messages for a stream on a SMS provider that is set to FireAndForget delivery and I'm trying to figure out why, because the message is delivered successfully to the target grain. I know this because the target grain will post the contents of the message to Discord and I get the messages from it.

There's only one consumer and many producers, and this message is logged from the producer.

```
2021-03-17T20:22:28.2593083+00:00 warn: Orleans.Runtime.CallbackData[100157]
      Response did not arrive on time in 00:00:30 for message: Request S172.31.45.129:11111:353706977*grn/Showdown.Backend.Grains.GameSessionGrain/36138d0f-f43e-4bcf-a5b6-d610335fd81f@dffaedb2->S172.31.45.129:11111:353706977*grn/Showdown.Backend.DiscordBot.ManagementDiscordBotGrain/0@cf270481 InvokeMethodRequest Orleans.Streams.IStreamConsumerExtension:DeliverImmutable #4432. Target History is: <S172.31.45.129:11111:353706977:*grn/Showdown.Backend.DiscordBot.ManagementDiscordBotGrain/0:@cf270481>. About to break its promise.
2021-03-17T20:22:28.2603819+00:00 warn: Orleans.Providers.Streams.SimpleMessageStream.SimpleMessageStreamProducer[0]
      => SpanId:e1559cd1cbde1643, TraceId:d407379882152148bb682bb7cd2b282e, ParentId:0000000000000000 => ConnectionId:0HM79FQTL0UML => RequestPath:/v1/Realtime/Server RequestId:0HM79FQTL0UML:00000002 => Showdown.Backend.Application.Controllers.RealtimeController.ServerWebsocketAsync (Showdown.Backend.Application) => GameSessionGrain 36138d0f-f43e-4bcf-a5b6-d610335fd81f OnEndPlay
      Failed to deliver message to consumer on c9ae96cc-a3e5-48fd-82bd-ab0fc143013a for stream SessionUpdate-00000000-0000-0000-0000-000000000000-SMSProvider.
      System.TimeoutException: Response did not arrive on time in 00:00:30 for message: Request S172.31.45.129:11111:353706977*grn/Showdown.Backend.Grains.GameSessionGrain/36138d0f-f43e-4bcf-a5b6-d610335fd81f@dffaedb2->S172.31.45.129:11111:353706977*grn/Showdown.Backend.DiscordBot.ManagementDiscordBotGrain/0@cf270481 InvokeMethodRequest Orleans.Streams.IStreamConsumerExtension:DeliverImmutable #4432. Target History is: <S172.31.45.129:11111:353706977:*grn/Showdown.Backend.DiscordBot.ManagementDiscordBotGrain/0:@cf270481>.
         at Orleans.Internal.OrleansTaskExtentions.<ToTypedTask>g__ConvertAsync|4_0[T](Task`1 asyncTask)
         at Orleans.Providers.Streams.SimpleMessageStream.SimpleMessageStreamProducerExtension.StreamConsumerExtensionCollection.DeliverToRemote(IStreamConsumerExtension remoteConsumer, StreamId streamId, GuidId subscriptionId, Object item, Boolean optimizeForImmutableData, Boolean fireAndForgetDelivery)
```

Here's the saved pub/sub state:

```json
{
  "$id": "1",
  "$type": "Orleans.Streams.PubSubGrainState, Orleans.Runtime",
  "Producers": {
    "$type": "System.Collections.Generic.HashSet`1[[Orleans.Streams.PubSubPublisherState, Orleans.Core]], System.Private.CoreLib",
    "$values": [
      {
        "$id": "2",
        "$type": "Orleans.Streams.PubSubPublisherState, Orleans.Core",
        "Stream": {
          "$id": "3",
          "$type": "Orleans.Streams.StreamId, Orleans.Core",
          "Guid": "00000000-0000-0000-0000-000000000000",
          "ProviderName": "SMSProvider",
          "Namespace": "SessionUpdate"
        },
        "producerReference": {
          "$type": "Orleans.Streams.OrleansCodeGenStreamProducerExtensionReference, Orleans.Core",
          "GrainId": "477021ba01c21a388dae569bc6460b9a03ffffff9a4f0d17",
          "GenericArguments": ""
        }
      }
    ]
  },
  "Consumers": {
    "$type": "System.Collections.Generic.HashSet`1[[Orleans.Streams.PubSubSubscriptionState, Orleans.Core]], System.Private.CoreLib",
    "$values": [
      {
        "$id": "4",
        "$type": "Orleans.Streams.PubSubSubscriptionState, Orleans.Core",
        "Stream": {
          "$id": "5",
          "$type": "Orleans.Streams.StreamId, Orleans.Core",
          "Guid": "00000000-0000-0000-0000-000000000000",
          "ProviderName": "SMSProvider",
          "Namespace": "SessionUpdate"
        },
        "SubscriptionId": {
          "$id": "6",
          "$type": "Orleans.Runtime.GuidId, Orleans.Core.Abstractions",
          "Guid": "c9ae96cc-a3e5-48fd-82bd-ab0fc143013a"
        },
        "consumerReference": {
          "$type": "Orleans.Streams.OrleansCodeGenStreamConsumerExtensionReference, Orleans.Core",
          "GrainId": "00000000000000000000000000000000030000001cb0d091",
          "GenericArguments": ""
        }
      }
    ]
  }
}
```

Any ideas on how I can troubleshoot this? There's only the single consumer and it's obviously getting the messages as I said, but I still get this exception in the logs for all the messages sent to this stream. It's not taking 30 seconds to deliver either - the `OnNextAsync` call is finishing in a few ms.

**Comment by highlyunavailable** [comment]
Turns out that this is because I'm sending the message (without waiting for confirmation) and then deactivating the grain. It then tries to deliver the confirmation back to the grain even though it's deactivated and gets a `NonExistentActivation` exception.

**Comment by ReubenBond** [comment]
This now fixed, as a part of #7231 - `NonExistentActivation` is a thing of the past. Many apologies for the slow response
